### ISSUE TITLE
Textarea tag content incorrectly indented (need per-element :ugly support)

### ISSUE THREAD

**Issue by jcoleman** [haml/haml#506]
See more detailed discussion on issue #503.

TL;DR: Rails 3.2.3 now emits a preceding newline in textarea tags to comply with the HTML spec (since browsers eat the first newline.) Haml incorrectly encodes that newline (fixed in #503.) The fix for that reveals another issue: since there is now an unencoded newline emitted, then if Haml is in `:ugly = false` mode, then the content of the textarea tag is incorrectly ideated leading to extra spaces in the displayed input box.

In indented mode, Haml needs to allow certain tag's additional lines to be unindented. There a few possible solutions:
1. If  possible (needs investigation) we could add a helper method that does this. Then we would add a text_area_tag wrapping monkey patch method to automatically call this helper when creating  textarea tag. For example:
   
   `= unindented(text_area_tag(...))`
2. Extend the language grammar to add _yet another_ preceding line character to denote this (e.g. we already have `=`, `-`, `~`.) For example: (HT lisp)
   
   `'= text_area_tag(...)`

What do you all think? Which path should we take? Anyone know if the first solution is even doable? Either way I'm assuming that we're looking at changes to the compiling code.
